public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -377,6 +377,10 @@ function transformToHistogramOverTime(seriesList: DataFrame[]) {
     for (let j = 0; j < topSeries.values.length; j++) {
       const bottomPoint = bottomSeries.values[j] || [0];
       topSeries.values[j] -= bottomPoint;
+
+      if (topSeries.values[j] < 1e-9) {
+        topSeries.values[j] = 0;
+      }
```

Comment:
think we can move this condition into an additional loop below these nested loops and only run this check & adjustment for the final (lowest) series?

e.g. if we have 1k values and 20 buckets you would be running this check 20k times (and we dont really need it in the middle of the de-accum stack). if you pull this out and only run it over the lowest series, then it will always run 1k times (or however many timestamps you got) no matter how many buckets you have.

Reply:
Sure thing! I'll make some updates.

Reply:
sorry, you're correct. negative buckets in the middle would also cause this, so we have to check all of them.